src/persistence.rs

```diff
@@ -52,11 +54,28 @@ impl vwmap::VwNamespaceMap {
     }
 }
 
+pub fn save_sharable_regressor_to_filename(
+    filename: &str,
+    mi: &model_instance::ModelInstance,
+    vwmap: &vwmap::VwNamespaceMap,
+    re: BoxedRegressorTrait,
+) -> Result<(), Box<dyn Error>> {
+    let output_bufwriter = &mut io::BufWriter::new(
+        fs::File::create(filename)
+            .expect(format!("Cannot open {} to save regressor to", filename).as_str()),
+    );
+    write_regressor_header(output_bufwriter)?;
+    vwmap.save_to_buf(output_bufwriter)?;
+    mi.save_to_buf(output_bufwriter)?;
+    re.write_weights_to_buf(output_bufwriter)?;
+    Ok(())
+}
+
 pub fn save_regressor_to_filename(
```

Comment:
looks like this is now only used for testing + saving inference weights, maybe we can get rid of it?

Reply:
Good question - not sure. I do think it's good to provide an API for a safe regressor, and I don't want to modify in this PR the code calling this function currently (will need to wrap it with `BoxedRegressorTrait`). So I think in this case I'll leave as is